tests/api/testcl.hpp

```diff
@@ -213,6 +213,13 @@ class WithContext : public ::testing::Test {
         ASSERT_CL_SUCCESS(err);
     }
 
+    void SetUpWithContextProperties(const cl_context_properties* _props) {
```

Comment:
Should be called by `WithContext::SetUp`

Reply:
`m_context` should also be released if not null, before creating a new context.
Maybe `m_context` should be initialized to null in the first place.

Reply:
I do not quite understand why `WithContext::SetUp` should call this function, can you please explain?

Reply:
We just want to avoid duplicating the code.
```
void SetUp() override {
  SetUpWithContextProperties(nullptr);
}
```